### PR TITLE
fix: add content_filter and function_call to finish_reason

### PR DESCRIPTION
**Pull request by rattrayalex**
Would address https://github.com/openai/openai-node/issues/320

I have not verified whether this or https://platform.openai.com/docs/guides/gpt/chat-completions-api is more accurate on this subject.

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -2509,7 +2509,7 @@ components:
                 description: &completion_finish_reason_description |
                   The reason the model stopped generating tokens. This will be `stop` if the model hit a natural stop point or a provided stop sequence,
                   or `length` if the maximum number of tokens specified in the request was reached.
```
Comment (rattrayalex) [inline openapi.yaml:2511]:
```suggestion
                  `length` if the maximum number of tokens specified in the request was reached,
                  `content_filter` if content was omitted due to a flag from our content filters,
                  or `function_call` if the model decided to call a function.

```

File: openapi.yaml [hunk 2]
```diff
@@ -2508,8 +2508,11 @@ components:
                 type: string
                 description: &completion_finish_reason_description |
                   The reason the model stopped generating tokens. This will be `stop` if the model hit a natural stop point or a provided stop sequence,
-                  or `length` if the maximum number of tokens specified in the request was reached.
-                enum: ["stop", "length"]
+                  `length` if the maximum number of tokens specified in the request was reached,
+                  `content_filter` if content was omitted due to a flag from our content filters,
+                  or `function_call` if the model decided to call a function.
+
+                enum: ["stop", "length", "content_filter", "function_call"]
```
Comment (athyuttamre) [inline openapi.yaml:2515]: This isn't quite right.

`v1/completions` only supports `stop`, `length`, and `content_filter`. `v1/chat/completions` supports all 4.

I think this line should be only the first three, and [this line](https://github.com/openai/openai-openapi/blob/90f86ecb3d4c4c7666b4cf7b5993692918aba5d3/openapi.yaml#L2820) should have all 4 instead.

Could you update? Thanks!

**Review by rattrayalex — COMMENTED** [review]
Oops, good catch @athyuttamre ! Got turned around. Re-review?

**Review by rattrayalex — COMMENTED** [review]
Oops, good catch @athyuttamre ! Got turned around. Re-review?

**Review by rattrayalex — COMMENTED** [review]
Oops, good catch @athyuttamre ! Got turned around. Re-review?

**Review by athyuttamre — APPROVED** [review]

**Pull request merged**